host-interaction/file-system/reference-absolute-stream-path-on-windows.yml

```diff
@@ -4,19 +4,23 @@ rule:
     namespace: host-interaction/file-system
     authors:
       - blas.kojusner@mandiant.com
+      - william.ballenthin@mandiant.com
     scope: basic block
     references:
       - https://learn.microsoft.com/en-us/windows/win32/fileio/file-streams
     examples:
       - 51828683DC26BFABD3994494099AE97D:0x11A9
   features:
     - and:
-      - string: /(..\?\\)?([\w]\:|\\)(\\((?![\<\>\"\/\|\*\?])[\x20-\x7E])+)+:(((?![\<\>\"\/\|\*\?])[\x20-\x7E])+)+$/
+      - string: /(..\?\\)?([\w]\:|\\)(\\+((?![\<\>\"\/\|\*\?\\])[\x20-\x5B\x5D-\x7E])+)+:(((?![\<\>\"\/\|\*\?\\])[\x20-\x5B\x5D-\x7E])+)+$/
```

Comment:
```suggestion
      - string: /(..\?\\)?([\w]\:|\\)(\\+((?![\<\>\"\/\|\*\?\:\\])[\x20-\x5B\x5D-\x7E])+)+(\:\$?([a-zA-Z0-9_]+))?$/
```

Reply:
so this enforces that the stream name must begin with "$"?

Reply:
The `$` is optional.

Reply:
ntfs-3g doesn't seem to require the `$` prefix:
https://linux.die.net/man/8/ntfsmount

Reply:
```ps1
PS C:\Users\william.ballenthin> set-content -stream "bar" -value "bar" .\a.txt
PS C:\Users\william.ballenthin> get-item .\a.txt -stream *


PSPath        : Microsoft.PowerShell.Core\FileSystem::C:\Users\william.ballenthin\a.txt::$DATA
PSParentPath  : Microsoft.PowerShell.Core\FileSystem::C:\Users\william.ballenthin
PSChildName   : a.txt::$DATA
PSDrive       : C
PSProvider    : Microsoft.PowerShell.Core\FileSystem
PSIsContainer : False
FileName      : C:\Users\william.ballenthin\a.txt
Stream        : :$DATA
Length        : 12

PSPath        : Microsoft.PowerShell.Core\FileSystem::C:\Users\william.ballenthin\a.txt:bar
PSParentPath  : Microsoft.PowerShell.Core\FileSystem::C:\Users\william.ballenthin
PSChildName   : a.txt:bar
PSDrive       : C
PSProvider    : Microsoft.PowerShell.Core\FileSystem
PSIsContainer : False
FileName      : C:\Users\william.ballenthin\a.txt
Stream        : bar
Length        : 5
```